internal/pkg/cli/task_run.go

```diff
@@ -945,6 +981,77 @@ func (o *runTaskOpts) deploy() error {
 	return o.deployer.DeployTask(os.Stderr, input, deployOpts...)
 }
 
+// deployEnvFileIfNeeded uploads the env file if needed, ensures that an S3 bucket is available, and returns the ARN of uploaded file.
+func (o *runTaskOpts) deployEnvFileIfNeeded() (string, error) {
+	if o.envFile == "" {
+		return "", nil
+	}
+
+	if o.envFileARN != "" {
+		return o.envFileARN, nil
+	}
+
+	region := aws.StringValue(o.sess.Config.Region)
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return "", fmt.Errorf("get application from store: %w", err)
+	}
+
+	// bootstrap S3 bucket
+	o.spinner.Start(fmt.Sprintf(fmtTaskRunResourcesStart, color.HighlightUserInput(o.appName)))
+	err = o.deployer.AddPipelineResourcesToApp(app, region)
```

Comment:
1. If the users are  running `copilot task run` in a Copilot application and environment with `--app` and `--env` won't the S3 bucket already exist? Can we reuse the existing environment bucket in that scenario?

2. It's possible that folks run `copilot task run` outside of a Copilot application with `--default` for example. In that scenario `AppPipelineResourcesToApp` will fail. 
Personally, I think I'm okay shipping this feature and restricting the `--env-file` flag to only work with `--app` and `--env` at first by returning an error in `Validate`. If anybody asks to expand it, we can add it on later.

Reply:
> If the users are running copilot task run in a Copilot application and environment with --app and --env won't the S3 bucket already exist? Can we reuse the existing environment bucket in that scenario?

I thought this was the case, but when I ran in a bare application without any services initialized I found that this was necessary to get things working.

I can also add the warning for that second case after testing tomorrow! 👍

Reply:
Thanks for taking a look here! The second point was indeed a potential source of bad UX.

1. I couldn't find any other app-wide or env-wide S3 buckets that get created other than the one created with the function that I call here.
2. I added some additional checks here as suggested, though I have to return the error in `Ask` instead:

```
PS C:\Users\rclinard\Projects\testing\magic-test> C:\Users\rclinard\Projects\copilot-cli\bin\local\copilot.exe task run --env-file .\magic.env --default
Couldn't find any applications in this region and account. Try initializing one with `copilot app init`
X no S3 bucket available to hold environment file .\magic.env specified in --env-file: ask for application: no apps found
```

Reply:
> I thought this was the case, but when I ran in a bare application without any services initialized I found that this was necessary to get things working.

I might be misunderstanding but after running `copilot env init` we call `cfn.AddEnvToApp` and a stackset instance is created with the `PipelineBuiltArtifactBucket` resource in it.

So we still shouldn't need to call `AddPipelineResourcesToApp` here if an env is initialized 🤔 .

Reply:
yup that's right!